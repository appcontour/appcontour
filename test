#!/bin/bash
#

l__ist=`ls tests/test.??`
if [ -n "$1" ]
then
  l__ist=tests/test.$1
fi

tmpf__ile=test$$.tmp
failed="0"

for f__ile in $l__ist
do
  i__nfo=$(grep "^#testname:" $f__ile | cut -f2- -d:)
  if [ -n "$i__nfo" ]; then i__nfo=" ($i__nfo)"; fi
  test=`echo $f__ile | cut -f2 -d\.`
  echo "Running test $test$i__nfo..."
  source $f__ile 2>/dev/null >$tmpf__ile
  exitcode=$?

  if [ -f ${f__ile}.out ]
  then
    if ! diff $tmpf__ile ${f__ile}.out
    then
      failed=$[ $failed + 1 ]
    fi
  else
    echo "...new test, copying out file"
    cp $tmpf__ile ${f__ile}.out
  fi
  rm $tmpf__ile
  echo "$exitcode" >$tmpf__ile
  if [ -f ${f__ile}.ec ]
  then
    if ! diff $tmpf__ile ${f__ile}.ec
    then
      failed=$[ $failed + 1 ]
    fi
  else
    echo "...new test, copying ec file"
    cp $tmpf__ile ${f__ile}.ec
  fi
  rm $tmpf__ile
done

if [ "$failed" -gt "0" ]
then
  echo $failed tests failed
  exit 1
fi

exit 0
