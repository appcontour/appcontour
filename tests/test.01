./contour applyrule n1 <examples/tre_sfere.morse
