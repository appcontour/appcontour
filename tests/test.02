#./contour applyrule n1 <examples/three_spheres.morse | ./contour applyrule n1 | ./contour applyrule n1 | ./contour applyrule n1
./contour applyrule k2 <examples/three_spheres.morse | ./contour applyrule k2 | ./contour applyrule k2 | ./contour applyrule k2
