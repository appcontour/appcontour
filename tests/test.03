./contour applyrule n1 <examples/tre_sfere.morse | ./contour applyrule n1 | ./contour applyrule n1
