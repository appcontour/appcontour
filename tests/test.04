./contour ishuffman -q examples/two_rings.morse
./contour iscontour -q examples/immersed/boysurface.morse
