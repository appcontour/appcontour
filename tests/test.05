./contour applyrule n2 <examples/sfera_dietro_toro.morse
./contour applyrule n3 <examples/sfera_davanti_toro.morse
./contour applyrule n4 <examples/due_anelli.morse | ./contour applyrule n2 | ./contour applyrule n3 | ./contour applyrule n1
./contour applyrule n4 <examples/toro2.morse
./contour applyrule N5 <examples/tre_sfere.morse | ./contour applyrule N5:6 | ./contour applyrule N1 | ./contour applyrule N1 | ./contour applyrule N1
