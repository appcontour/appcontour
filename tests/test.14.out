Region 0:
Region 1:
-r 1 --stratum 0 (INVL:1)
-r 1 --stratum 1 (INVL:2)
-r 1 --stratum 2 (INVL:3)
-r 1 --stratum 3 (INVL:4)
Region 2:
-r 2 --stratum 0 (INVL:5)
-r 2 --stratum 1 (INVL:6)
-r 2 --stratum 2 (INVL:7)
-r 2 --stratum 3 (INVL:8)
Region 3:
Region 4:
-r 4 --stratum 0 (INVL:9)
-r 4 --stratum 1 (INVL:10)
sketch {
Arc 3: (0 1 0);
Arc 1: (0);
Arc 2: (0);
Region 0 (f = 0): () (-a1) (-a2);
Region 3 (f = 0): (+a3);
Region 1 (f = 2): (+a1) (-a3);
Region 2 (f = 2): (+a2);
}
sketch {
Arc 6: (1 2 1);
Arc 1: (0);
Arc 2: [0 1 2);
Arc 3: [2 1 0);
Arc 4: [0];
Arc 5: [0];
Region 0 (f = 0): () (-a1);
Region 5 (f = 0): (+a6);
Region 1 (f = 4): (+a2);
Region 2 (f = 4): (+a3);
Region 3 (f = 0): (-a4 -a5);
Region 4 (f = 2): (+a1) (-a6) (-a2 +a4 -a3 +a5);
}
sketch {
Arc 1: (0);
Arc 2: (0 1 0);
Arc 3: [0 1 2);
Arc 4: [2 1 0);
Arc 5: [0];
Arc 6: [0];
Region 0 (f = 0): () (-a1);
Region 1 (f = 6): (+a2);
Region 2 (f = 4): (+a4);
Region 3 (f = 0): (-a5 -a6);
Region 4 (f = 2): (+a1) (-a3 +a6 -a4 +a5);
Region 5 (f = 4): (+a3) (-a2);
}
